client/src/FiltersPage.js

```diff
@@ -0,0 +1,181 @@
+import './App.css';
+import Login from './Login';
+import CardComponent from './CardComponent';
+import Filters from './Filters';
+import NavBar from './navbar';
+import React from 'react';
+import './Filter.css';
+
+export default function FiltersPage() {
+  ///////////////////////////////////////////////////////////////////////////////////
+  ///////////////////////////////////////////////////////////////////////////////////
+  // Map and List view mode buttons
+
+  const [mode, setMode] = React.useState(1);
+
+  function render() {
+    if (mode == 1) {
+      return (
+        <div>
+          <CardComponent
+            name="Bianca's Radishes"
+            location="Berkeley, CA"
+            tags={['vegetables', 'fruit']}
+          />
+          <CardComponent
+            name="Bianca's Radishes"
+            location="Berkeley, CA"
+            tags={['vegetables', 'fruit']}
+          />
+          <CardComponent
+            name="Bianca's Radishes"
+            location="Berkeley, CA"
+            tags={['vegetables', 'fruit']}
+          />
+          <CardComponent
+            name="Bianca's Radishes"
+            location="Berkeley, CA"
+            tags={['vegetables', 'fruit']}
+          />
+          <CardComponent
+            name="Bianca's Radishes"
+            location="Berkeley, CA"
+            tags={['vegetables', 'fruit']}
+          />
+          <CardComponent
+            name="Bianca's Radishes"
+            location="Berkeley, CA"
+            tags={['vegetables', 'fruit']}
+          />
+          <CardComponent
+            name="Bianca's Radishes"
+            location="Berkeley, CA"
+            tags={['vegetables', 'fruit']}
+          />
+        </div>
+      );
+    } else {
+      return <div>map please </div>;
+    }
+  }
+
+  function list_button() {
+    if (mode == 1) {
+      return (
+        <button
+          className="activeButton"
+          type="button"
+          onClick={() => setMode(1)}
+        >
+          List
+        </button>
+      );
+    } else {
+      return (
+        <button
+          className="inactiveButton"
+          type="button"
+          onClick={() => setMode(1)}
+        >
+          List
+        </button>
+      );
+    }
+  }
+
+  function map_button() {
+    if (mode == 2) {
+      return (
+        <button
+          className="activeButton"
+          type="button"
```

Comment:
notice how there's a lot of shared code between these two functions (and the only thing different is the class names, button text, and `setMode` input, which means you can replace this with a single function.

nit: to improve code cleanliness, you can see that the only difference between the two buttons you return from the `if/else` is the class name; so you can shorten this with a ternary operator and replace with
```javascript
function renderToggleButton(buttonText, activeOnListMode) {
      return (
        <button
          className={activeOnListMode === listMode ? "activeButton" : "inactiveButton"}
          type="button"
          onClick={() => setListMode(!listMode)}
        >
          {buttonText}
        </button>
      );
}
```

Then you can use `renderToggleButton('List', true)` and `renderToggleButton('Map', false)` to render your List and Map toggle buttons.